test/Spago/Build/Polyrepo.purs

```diff
@@ -0,0 +1,349 @@
+module Test.Spago.Build.Polyrepo where
+
+import Test.Prelude
+
+import Data.Array as Array
+import Data.String (Pattern(..))
+import Data.String as String
+import Node.FS.Aff as FSA
+import Node.Path as Path
+import Registry.Version as Version
+import Spago.Command.Init (DefaultConfigOptions(..))
+import Spago.Command.Init as Init
+import Spago.Core.Config as Config
+import Spago.FS as FS
+import Test.Spec (SpecT)
+import Test.Spec as Spec
+import Test.Spec.Assertions as Assert
+
```

Comment:
Might be worth linking this PR here in this module, so that visitors from the future will be able to find the diagrams in the PR description, as understanding the various cases with those is easier

Reply:
What happens if we add more tests here? Do we link to those PRs as well?

I think I'd rather copy-paste the Mermaid diagram into this module's docs. If we add a new test to this situation, we can update the diagrams as well.

Reply:
Ah yes good point. Let's go with adding the mermaid diagrams here.